example/lib/encrypted_messaging_service.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:ably_flutter/ably_flutter.dart' as ably;
+import 'package:crypto/crypto.dart';
+import 'package:rxdart/rxdart.dart';
+
+class EncryptedMessagingService {
+  static const channelName = 'encrypted-test-channel';
+  final ably.Realtime? _realtime;
+  ably.Rest? _rest;
+  ably.RealtimeChannel? _realtimeChannel;
+  ably.RestChannel? _restChannel;
+
+  final BehaviorSubject<List<ably.Message>> messageHistoryBehaviorSubject =
+      BehaviorSubject<List<ably.Message>>.seeded([]);
+
+  ValueStream<List<ably.Message>> get messageHistoryStream =>
+      messageHistoryBehaviorSubject.stream;
+
+  static const examplePassword = 'password-to-encrypt-and-decrypt-text';
+
+  Uint8List get keyFromPassword {
```

Comment:
Whilst this is only an example, it's worth stating explicitly that this is not the recommended way to derive a key from a password, production code should use a KDF or some other secure, attack-resistant mechanism.

Reply:
Thanks. I've done it in 2b6e91b0928251dbcc8f49f916023735c3c57cbc